src/ChainSafe.Gaming.MetaMask.Unity/MetaMaskController.cs

```diff
@@ -0,0 +1,197 @@
+using System.Collections;
+using System.Threading.Tasks;
+using ChainSafe.Gaming.Web3;
+using ChainSafe.Gaming.Web3.Environment;
+using Nethereum.Hex.HexTypes;
+using Nethereum.JsonRpc.Client;
+using Nethereum.Unity.Metamask;
+using Nethereum.Unity.Rpc;
+using UnityEngine;
+
+namespace ChainSafe.Gaming.MetaMask.Unity
+{
+    /// <summary>
+    /// Controller script for connecting to MetaMask and making awaitable JsonRPC requests.
+    /// </summary>
+    public class MetaMaskController : MonoBehaviour
```

Comment:
Can we make it a simple plain C# object instead of MonoBehaviour?

Reply:
Nope, the plugin needs `gameObject` name to interact with the script, It uses reflection

Reply:
I also tried passing it as an argument but it doesn't work if the Callbacks are on a different script than what's on the `gameObject`